addons/point_of_sale/models/stock_picking.py

```diff
@@ -92,7 +92,7 @@ def _create_move_from_pos_order_lines(self, lines):
             current_move = self.env['stock.move'].create(
                 self._prepare_stock_move_vals(first_line, order_lines)
             )
-            confirmed_moves = current_move._action_confirm()
```

Comment:
why not merging them? should not have any impact as you cannot physically distinguish them

Reply:
In my comment, you will see the 3rd issue:

> 3. Sell a kit and a component with invoice
    - https://drive.google.com/file/d/1rNRBYGorUCxzgqiIWO2hjZR6o_pS1jEy/view
    - The kit has expense of `1.0`, wrong.
    - The component has expense of `80.0`, correct.

The reason why the kit costs `1.0` and not `81.0` is because when selling a Wood Panel and a Table Kit, the kit generates stock moves for its component (1 Wood Panel and 4 Bolts). And once the 'Wood Panel' line is processed, it will merge itself with the move of the kit components. So, there will only be 2 stock moves (2 Wood Panel and 4 Bolts). The new method `_get_anglo_saxon_moves` will returned insufficient moves for the `_compute_average_price`, resulting to wrong unit price.

Reply:
Why are they merge if we have "bom_line_id" on a move and not on the other?

Reply:
That I didn't investigate. I was also wondering why the moves are merged and then when I tried to debug, I saw the optional param `merge` of `_action_confirm`.